### PR TITLE
build_docker: Add `description` and `summary` labels

### PR DESCRIPTION
**Pull request by yaronkaikov**
Adding description and summary labels to our docker images per @tzach and @mykaul request,

Related to https://github.com/scylladb/scylla-enterprise/issues/863

### CODE REVIEW

**Review by tzach — APPROVED** [review]
LGTM

**Comment by yaronkaikov** [comment]
Locally build also enterprise docker image 
```
          "Labels": {
               "description": "Customer: download and use Scylla Enterprise Docker. Non Customers: try it free for 30 days.",
               "io.buildah.version": "1.33.2",
               "org.opencontainers.image.ref.name": "ubuntu",
               "org.opencontainers.image.version": "22.04",
               "summary": "ScyllaDB Enterprise"
          },
```
@mykaul @tzach

**Comment by mykaul** [comment]
> "description": "Customer: download and use Scylla Enterprise Docker. Non Customers: try it free for 30 days.",

I'd replace the word 'Docker' with 'Container'

**Comment by mykaul** [comment]
> "org.opencontainers.image.ref.name": "ubuntu",
>                "org.opencontainers.image.version": "22.04",

I'd remove both.

**Comment by yaronkaikov** [comment]
> > "org.opencontainers.image.ref.name": "ubuntu",
> > "org.opencontainers.image.version": "22.04",
> 
> I'd remove both.

```
          "Labels": {
               "description": "",
               "io.buildah.version": "1.33.2",
               "org.opencontainers.image.ref.name": "",
               "org.opencontainers.image.version": "",
               "summary": "ScyllaDB Open Source"
          },

```

**Comment by yaronkaikov** [comment]
> > > "org.opencontainers.image.ref.name": "ubuntu",
> > > "org.opencontainers.image.version": "22.04",
> > 
> > 
> > I'd remove both.
> 
> ```
>           "Labels": {
>                "description": "",
>                "io.buildah.version": "1.33.2",
>                "org.opencontainers.image.ref.name": "",
>                "org.opencontainers.image.version": "",
>                "summary": "ScyllaDB Open Source"
>           },
> ```

@mykaul Removed

File: dist/docker/debian/build_docker.sh [hunk 1]
```diff
@@ -22,6 +22,13 @@ if uname -m | grep aarch64 ; then
   arch="arm64"
 fi
 
+if [ "$product" = "scylla" ]; then
+  summary_label="ScyllaDB Open Source"
+  description_label=""
```
Comment (avikivity) [inline dist/docker/debian/build_docker.sh:27]: Keep just this part and move the enterprise part to the enterprise repo.

Add a description here.

File: dist/docker/debian/build_docker.sh [hunk 2]
```diff
@@ -102,6 +109,10 @@ bconfig --entrypoint  '["/docker-entrypoint.py"]'
 bconfig --cmd  ''
 bconfig --port 10000 --port 9042 --port 9160 --port 9180 --port 7000 --port 7001 --port 22
 bconfig --volume "/var/lib/scylla"
+bconfig --label org.opencontainers.image.ref.name
+bconfig --label org.opencontainers.image.version
```
Comment (avikivity) [inline dist/docker/debian/build_docker.sh:113]: What does this do?

Comment (mykaul) [reply dist/docker/debian/build_docker.sh:113]: It removes those default ones.

**Comment by mykaul** [comment]
Finally found the complete spec for annotations - https://github.com/opencontainers/image-spec/blob/main/annotations.md  - need to see that we are compliant.

File: dist/docker/debian/build_docker.sh (hunk 2 above)
Comment (yaronkaikov) [reply dist/docker/debian/build_docker.sh:113]: Add an empty string, the default is 
```
"org.opencontainers.image.ref.name": "ubuntu",
"org.opencontainers.image.version": "22.04",
```
@mykaul asked to remove those

File: dist/docker/debian/build_docker.sh (hunk 1 above)
Comment (yaronkaikov) [reply dist/docker/debian/build_docker.sh:27]: Do you mean get rid of the `if` and open another PR under enterprise repo?

File: dist/docker/debian/build_docker.sh (hunk 2 above)
Comment (avikivity) [reply dist/docker/debian/build_docker.sh:113]: shouldn't we put our own instead?

Comment (mykaul) [reply dist/docker/debian/build_docker.sh:106]: We should.

File: dist/docker/debian/build_docker.sh [hunk 3]
```diff
@@ -102,6 +102,10 @@ bconfig --entrypoint  '["/docker-entrypoint.py"]'
...
+bconfig --label description="ScyllaDB Open Source"
```
Comment (mykaul) [inline dist/docker/debian/build_docker.sh:107]: I expect the description to be different than the summary.
Example for a description:
```
NoSQL data store using the seastar framework, compatible with Apache Cassandra```

Comment (yaronkaikov) [reply dist/docker/debian/build_docker.sh:107]: Changed, for enterprise we have different as @tzach asked. should we add both?

File: dist/docker/debian/build_docker.sh (hunk 2 above)
Comment (yaronkaikov) [reply dist/docker/debian/build_docker.sh:106]: What should we have instead?

Comment (mykaul) [reply dist/docker/debian/build_docker.sh:106]: Version is easy - we know what the version is (I'd add the complete version, inc. build ID, etc.

Comment (mykaul) [reply dist/docker/debian/build_docker.sh:106]: Name could be just 'ScyllaDB'

Comment (yaronkaikov) [reply dist/docker/debian/build_docker.sh]:
```
          "Labels": {
               "description": "ScyllaDB Open Source",
               "io.buildah.version": "1.33.2",
               "org.opencontainers.image.ref.name": "ScyllaDB",
               "org.opencontainers.image.version": "5.5.0~dev-0.20231221.3b5ae5101b23",
               "summary": "NoSQL data store using the seastar framework, compatible with Apache Cassandra"
          },
```

Changed

**Comment by yaronkaikov** [comment]
@mykaul @avikivity Any other comments?

**Review by mykaul — APPROVED** [review]

**Comment by yaronkaikov** [comment]
@scylladb/scylla-maint Can you please merge?

**Pull request closed without merging**